.scripts/pre-commit

```diff
@@ -1,30 +1,24 @@
 #!/bin/bash
 
-GREEN='\033[0;32m'
-RED='\033[0;31m'
-NO_COLOR='\033[0m'
+RED=$(printf '\033[0;31m')
+CLEAR=$(printf '\033[0m')
 
-echo "*********************************************************"
-echo "Running git pre-commit hook. Running spotlessApply and Git Add... "
-echo "---------------------------------------------------------"
-
-# Run spotless apply
-./gradlew spotlessApply
+# Find all files to be committed
+readarray -d '' changes -u <(git diff -z --staged --name-only)
+# Run spotless apply on changed files
+./gradlew spotlessApply "${changes[@]}"
 
 status=$?
 
-if [ "$status" = 0 ] ; then
-    echo -e "${GREEN}Static analysis found no problems. Running add${NO_COLOR}"
-    git add .
+if [ "$status" = 0 ]; then
+    echo "Static analysis succeeded. Updating commit."
```

Comment:
return the green

Reply:
My reasoning was that a successful run should not stand out when committing, so you can focus only on anything going wrong

Reply:
I prefer it to be easy to see good and bad runs (honesty i really like colors)

Reply:
agree with yoav